internal/model/matcher.go

```diff
@@ -22,3 +26,29 @@ func (c CompositePathMatcher) Matches(f string, isDir bool) (bool, error) {
 }
 
 var _ PathMatcher = CompositePathMatcher{}
+
+type singlePathMatcher struct {
```

Comment:
there's probably a better place to put this but I put it here for now to avoid circular dependency issues.